packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { isNullOrUndefined } from 'util';
+import * as vscode from 'vscode';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '../commands';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+
+export class ForceListMetadataExecutor extends SfdxCommandletExecutor<string> {
+  private metadataType: string;
+  private outputPath: string;
+  private defaultUsernameOrAlias: string;
+
+  public constructor(
+    metadataType: string,
+    outputPath: string,
+    defaultUsernameOrAlias: string
+  ) {
+    super();
+    this.metadataType = metadataType;
+    this.outputPath = outputPath;
+    this.defaultUsernameOrAlias = defaultUsernameOrAlias;
+  }
+
+  public build(data: {}): Command {
+    let builder = new SfdxCommandBuilder()
+      .withArg('force:mdapi:listmetadata')
+      .withFlag('-m', this.metadataType)
+      .withFlag('-u', this.defaultUsernameOrAlias)
+      .withFlag('-f', this.outputPath)
+      .withJson();
+
+    if (
+      this.metadataType === 'Dashboard' ||
+      this.metadataType === 'Report' ||
+      this.metadataType === 'EmailTemplate' ||
+      this.metadataType === 'Document'
+    ) {
+      builder = builder.withFlag('--folder', 'unfiled$public');
```

Comment:
There's an issue with this, only EmailTemplate and Report types have an out-of-the-box public folder called `unfiled$public`.
Overall how are we thinking about presenting this to the user ? Is the user going to see a top-level option in the tree-view called 'Report', then see only the reports under `unfiled$public` when they expand it ?

Reply:
Overall, we were thinking of having a top-level option ie. `Report` that would load the reports from `unfiled$public` and the custom folders that contain the rest of the reports. Loading the custom folders under these types will be done by the work item @brpowell is on. 
We were under the impression that the `unfiled$public` folder exists for all four types, Bryan and I will sync on how to handle this now

Reply:
For now, we'll just list the `unfiled$public` data for EmailTemplate and Report. Dashboards and Documents will just display the other folders under them when expanded.